### PR TITLE
Feat: Create multiple files in one single commit

### PR DESCRIPTION
**Pull request by daniel-hutao**
Signed-off-by: Daniel Hu <tao.hu@merico.dev>

# Summary

Feat: Create multiple files in one single commit

see https://github.com/daniel-hutao/golang-demo for more info

### Screenshots
image

### CODE REVIEW

**Comment by daniel-hutao** [comment]
image

File: internal/pkg/plugin/reposcaffolding/github/golang/golang.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ const (
 	DefaultWorkPath      = ".github-repo-scaffolding-golang"
 	DefaultTemplateRepo  = "dtm-scaffolding-golang"
 	DefaultTemplateOwner = "merico-dev"
+	TransBranch          = "init-with-devstream"
```
Comment (IronCore864) [inline internal/pkg/plugin/reposcaffolding/github/golang/golang.go:21]: "Trans" as a prefix can mean many things, transform, transaction, translate, etc. I think we can use a longer full name here to be perfectly clear.

File: internal/pkg/plugin/reposcaffolding/github/golang/golang.go [hunk 2]
```diff
@@ -90,7 +93,17 @@ func WalkLocalRepoPath(repoPath string, param *Param, ghClient *github.Client) e
 		if newPathForGithub, err := replaceAppNameInPathStr(pathForGithub, appName); err != nil {
 			return err
 		} else {
-			return ghClient.CreateFile(content, strings.TrimSuffix(newPathForGithub, ".tpl"))
+			// the main branch needed a initial commit
```
Comment (IronCore864) [inline internal/pkg/plugin/reposcaffolding/github/golang/golang.go:96]: "needed" -> "needs". It's generic, so using the present tense.

File: internal/pkg/plugin/reposcaffolding/github/golang/golang.go [hunk 3]
```diff
@@ -90,7 +93,17 @@ func WalkLocalRepoPath(repoPath string, param *Param, ghClient *github.Client) e
...
+			if strings.Contains(newPathForGithub, "gitignore") {
+				err := ghClient.CreateFile(content, strings.TrimSuffix(newPathForGithub, ".tpl"), MainBranch)
+				if err != nil {
+					log.Debugf("Failed to Add the .gitignore file.")
```
Comment (IronCore864) [inline internal/pkg/plugin/reposcaffolding/github/golang/golang.go:100]: "Add" -> "add", remove uppercase "A"

File: pkg/util/github/commit.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import (
+	"fmt"
+
+	"github.com/google/go-github/v42/github"
+
+	"github.com/merico-dev/stream/internal/pkg/log"
+)
+
+func (c *Client) GetLastCommit() (*github.RepositoryCommit, error) {
```
Comment (IronCore864) [inline pkg/util/github/commit.go:11]: Where is this function used?

**Review by IronCore864 — CHANGES_REQUESTED** [review]

**Comment by IronCore864** [comment]
No more comments.

File: pkg/util/github/pullrequest.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+	return pr.GetNumber(), nil
+}
+
+func (c *Client) MergePullRequest(number int, mergeMethod MergeMethod) error {
+	commitMsg := "Scaffolding with DevStream"
+	ret, _, err := c.PullRequests.Merge(c.Context, c.Owner, c.Repo, number, commitMsg, &github.PullRequestOptions{
+		CommitTitle: commitMsg,
+		SHA:         "",
+		// "merge", "squash", and "rebase"
```
Comment (lfbdev) [inline pkg/util/github/pullrequest.go:50]: remove this statement

File: internal/pkg/plugin/reposcaffolding/github/golang/golang.go (hunk 1 above)
Comment (daniel-hutao) [reply internal/pkg/plugin/reposcaffolding/github/golang/golang.go:21]: ok

File: pkg/util/github/pullrequest.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+		MergeMethod:        string(mergeMethod),
+		DontDefaultIfBlank: false,
+	})
+	if err != nil {
+		log.Debugf("Got some error when merge the pr: %s", err)
```
Comment (lfbdev) [inline pkg/util/github/pullrequest.go:55]: maybe ‘errors’ is better？

File: internal/pkg/plugin/reposcaffolding/github/golang/golang.go (hunk 1 above)
Comment (daniel-hutao) [reply internal/pkg/plugin/reposcaffolding/github/golang/golang.go]: TransitBranch is better here

File: pkg/util/github/commit.go (hunk 4 above)
Comment (daniel-hutao) [reply pkg/util/github/commit.go:11]: It's not used now. But it's ok left here as a util function at `util/github` package.

File: pkg/util/github/pullrequest.go (hunk 6 above)
Comment (daniel-hutao) [reply pkg/util/github/pullrequest.go:55]: or `Got a error ...`

Comment (daniel-hutao) [reply pkg/util/github/pullrequest.go]: an error

Comment (lfbdev) [reply pkg/util/github/pullrequest.go]: Got an error

**Review by lfbdev — APPROVED** [review]

**Pull request merged**